cyclepainter.py

```diff
@@ -389,17 +458,20 @@ def draw_basics(self):
                         c='k', marker='x', zorder=3)
 
         # draw kappa circles around branch points
+        # kappa-circles are added as patches such that the autoscaling of axes recognises them, see https://github.com/matplotlib/matplotlib/issues/2202/.
+        # Maybe this will cause errors later?
         for x in self.branch_points:
             if x.is_finite:
-                self.ax.add_artist(plt.Circle((x.real, x.imag), self.kappa, color='k', alpha=0.06, linestyle=':'))
+                self.ax.add_patch(plt.Circle((x.real, x.imag), self.kappa, color='k', alpha=0.06, linestyle=':'))
 
         # draw discriminant points
         self.ax.scatter([x.real() for x in self.surface.discriminant_points],
                         [x.imag() for x in self.surface.discriminant_points],
                         c='k', marker='.', zorder=3)
 
         # annotate all of the branch points
-        d_ann = (min(self.imag_span, self.real_span)/100).n()
+        # For some reason the .n() method is not reliable?
+        d_ann = numerical_approx(min(self.imag_span, self.real_span)/100)
```

Comment:
Maybe I'm missing something, but where is `numerical_approx` defined?

Reply:
numerical_approx is a method built into Sage, I do not know when but certainly since 8.9 (which is the version I have used in all the current testing).